### DISCUSSION TITLE
ProjectTo ArgumentException "Argument types do not match" with dictionaries

### DISCUSSION THREAD

**Discussion by DaanWasscher** [LuckyPennySoftware/AutoMapper#3801] [NA]
https://gist.github.com/DaanWasscher/6a44c6843c6215f490e6c543108e0424

AutoMapper 10.1.1
net6.0

When the source and destination have a dictionary property for which the values need to be mapped it won't map with ProjectTo.

Calling Map results in a mapped result, all good.
Calling ProjectTo results in an argument exception from AutoMapper.

**Comment by lbargaoanu** [comment]
I don't know what EF provider supports that, but EF6 certainly doesn't.

**Reply by DaanWasscher** [reply]
In EF Core 6 entity configuration I have a [valueconverter](https://docs.microsoft.com/en-us/ef/core/modeling/value-conversions?tabs=data-annotations) configured on the dictionary property that converts the dictionary to a json string when writing to database and back to dictionary upon reading from database.

I assumed that with EF Core's guidance for value-conversion in use that the projection with AutoMapper would work.

**Reply by lbargaoanu** [reply]
Drop all value converters and use `Map`. See https://github.com/AutoMapper/AutoMapper/discussions/3763 for a similar case.

**Reply by DaanWasscher** [reply]
#3763 is about valueconversion in the mapping configuration. I use valueconversion inside ef core. Or do you see those as the same thing?

**Reply by lbargaoanu** [reply]
Pretty much, but he also used EF Core converters. It's the same situation and the same solution applies.
